lib/data/file/backend.js

```diff
@@ -65,11 +74,15 @@ export const backend = {
                 return callback(errors.InternalError);
             })
             .on('end', () => {
-                log.debug('finished writing data', { key });
-                fs.fsync(fd, () => {
-                    fs.close(fd);
-                    return callback(null, key);
-                });
+                streamEnd = true;
+                // Only in flowing mode when fs.write finished
+                if (request._readableState.flowing) {
```

Comment:
@LaurenSpiegel yes you right and this is why i was thinking about it . I was thinking that the purpose was to wait the ```end``` of the readable stream .
So instead of putting a variable waiting for the end of the readable stream who i think doesn't mean that all datas have been called at the next call of ```on.('data')```.   Even with the ```resume``` witch is asynchronous.
Why not just listen ```finish``` on the writable stream and ```sync``` when all the data have been write ?